### PR TITLE
Replaced deprecated io/ioutil functions

### PR DESCRIPTION
**Pull request by slab713**
**What type of PR is this?** --> cleanup

**Which issue does this PR fix** --> io/ioutil package is deprecated 

**What does this PR do / Why do we need it** --> using recommended os & io packages

Issue #767 ... just updated a few files

**Will this break upgrades or downgrades?** --> i don't think so

**Does this PR introduce any user-facing change?** --> i don't think so

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rchincha** [comment]
Pls. also add this line to the PR description above:

`By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.`

**Review by rchamarthy — DISMISSED** [review]

**Comment by andaaron** [comment]
Hi @slab713 , besides fixing the build/test failures, please sign-off and gpg sign the commit - see the git commit parameters below. See this article for GPG signature verification details: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification

`git commit -s -S`

File: pkg/extensions/sync/sync_test.go [hunk 1]
```diff
@@ -3680,7 +3680,7 @@ func TestSyncSignaturesDiff(t *testing.T) {
 		So(err, ShouldBeNil)
 
 		defer func() { _ = os.Chdir(cwd) }()
-		tdir, err := ioutil.TempDir("", "sigs")
+		tdir, err := os.MkdirTemp(", "sigs")
```
Comment (andaaron) [inline pkg/extensions/sync/sync_test.go:3683]: Please fix this typo.

Also please remove ioutil from imports in this file.

File: pkg/extensions/sync/sync_test.go [hunk 2]
```diff
@@ -3773,7 +3773,7 @@ func TestSyncSignaturesDiff(t *testing.T) {
 
 		// now add new signatures to upstream and let sync detect that upstream signatures changed and pull them
 		So(os.RemoveAll(tdir), ShouldBeNil)
-		tdir, err = ioutil.TempDir("", "sigs")
+		tdir, err = os.MkdirTemp(", "sigs")
```
Comment (andaaron) [inline pkg/extensions/sync/sync_test.go:3776]: Please fix this typo

Comment (slab713) [reply pkg/extensions/sync/sync_test.go]: Done

File: pkg/extensions/sync/sync_test.go (hunk 1 above)
Comment (slab713) [reply pkg/extensions/sync/sync_test.go]: Done

**Comment by andaaron** [comment]
In y

> Added missing imports in a few files and fixed 2x missing (
> 
> @andaaron I connected a GPG key and ran `git commit -s -S`
> 
> Can you tell me more about the squash part? There are different opinions how to do it etc. Don't want to mess up something.

Well, we usually use `git rebase -i` and from the editor use the s option to merge the commit on a specific line in the commit on the previous line.

Steps should be below (assuming you have a git remote called `project-zot` pointing to the source repo at `project-zot/zot` and a git remote called `origin` pointing at repo `slab713/zot`). You can check your remotes using the `git remote -v` command, there are also commands for adding a remote if one of the above is missing.

Just to make sure your local main is up to date with project/zot:main and that merge commit you added from github UI is cleaned up - as we generally want linear git history.

```
git checkout main # make sure you are on main branch, which you have used for development so far
git pull --rebase project-zot main
git log
```

After this, you should only have 7 commits on top of what is on project-zot/zot:main, as the merge commit should be gone.
```
git rebase -i HEAD~7 -s -S
```

This will open an editor for you to rewrite history for the has 7 commits on the current branch - you can decide what to do with individual commits
On the 1st column, you will leave the `pick` keyword for your 1st commit on the 1st line. While the rest of the 6 commits on the next lines should have the `squash` keyword on the 1st column
After you save in the editor and close it, it should open a new editor instance for you to write the new commit message for the 1st and now only commit, this will be the single commit message for all your work on this PR
After you save in the editor and exit again you are back in the command line
If you look at the git log you should see just 1 commit with your modified message, which contains all your changes.

Force push and rewrite history on slab713/zot
```
git log
git push -f 
```

It would be easier next time if you use `git commit --amend` to update your previous commit instead of creating new ones :)
Also, it would be better when working on a PR to create a new branch for development, instead of committing to your `main`. It's easier to keep `main` to date if someone merges something to it after you have already opened the PR.

File: pkg/api/routes.go [hunk 3]
```diff
@@ -37,6 +36,7 @@ import (
 	localCtx "zotregistry.io/zot/pkg/requestcontext"
 	"zotregistry.io/zot/pkg/storage"
 	"zotregistry.io/zot/pkg/test" // nolint:goimports
+
```
[the remark this answers is not in the document]
Comment (andaaron) [reply pkg/api/routes.go]: How did you build locally if the make command does not work? :D
This is the make I am referring to https://www.gnu.org/software/make/
Depending on your operating system there are different ways to install it.

**Comment by slab713** [comment]
Okay, sorry for the hassle, there has been error after error. Thank you very much for the explanations. The commits appear to be signed now. @andaaron Yeah, Windows, Git Bash & VS Code. Got greeted with over 30 problems after cloning. They're gone now, except those three: https://i.imgur.com/s2paaqY.png

Also, I only see my own fork when using `git remote -v`, no pointer to the original repository.

**Review by andaaron — APPROVED** [review]

**Pull request merged**